[FIX] rma: portal views access errors

chienandalu opened the pull request:
- Portal mail thread needs token config.
- Unpublished products will raise AccessError on RMAs portal views for
portal users due to record rules.
- Set split window action domain to avoid context weird behavior

cc @Tecnativa TT24986

chienandalu committed:
[FIX] rma: portal views access errors

- Portal mail thread needs token config.
- Unpublished products will raise AccessError on RMAs portal views for
portal users due to record rules.
- Ensure active_id when getting actions in rma, since we could come from
a context that pollutes the expected active rma id.

pedrobaeza commented on rma/views/rma_portal_templates.xml:
```diff
@@ -262,6 +269,10 @@
                 <t t-call="portal.message_thread">
                     <t t-set="object" t-value="rma"/>
+                    <t t-set="object" t-value="rma"/>
+                    <t t-set="token" t-value="rma.access_token"/>
```
Do we have "Share" button in the backend?

chienandalu replied:
Yes

kos94ok-3D replied:
duplicated line `<t t-set="object" t-value="rma"/>`

chienandalu replied:
Fixed. Thanks @kos94ok-3D

pedrobaeza commented on rma/views/rma_portal_templates.xml:
```diff
@@ -132,12 +133,20 @@
                                     <div class="col-12 col-sm-8">
-                                        <span t-field="rma.product_id"/>
+                                        <!-- Don't depend on website -->
+                                        <t t-if="'website_published' in rma.sudo().product_id._fields and rma.sudo().product_id.website_published">
```
Why not simply put the regular one? How does `sale` works with this?

chienandalu replied:
they put line.name, wich isn't valid here

pedrobaeza replied:
OK, but I think you should simply avoid the website_published stuff. Why `t-field` is not valid in every case?

chienandalu replied:
`website` adds this record rule: https://github.com/odoo/odoo/blob/12.0/addons/website/security/website_security.xml#L56-L61 So if the portal tries to access those records, it will raise an AcessError.

Another option to avoid this would be to create a compute field with the product name

pedrobaeza replied:
Well, you have put website.page record rule, which doesn't apply here, but I suppose you mean website_sale record rule over products. Let's put directly the t-esc over display name with the sudoed record

chienandalu replied:
Perfect. I let it like that

pedrobaeza commented on rma/views/rma_portal_templates.xml:
```diff
@@ -132,12 +133,20 @@
+                                            <span t-field="rma.product_id"/>
+                                        </t>
+                                        <t t-else="">
+                                            <span t-esc="rma.sudo().product_id.name"/>
```
It must be `display_name`

chienandalu commented:
Changes done. I detected another issue with portal pickings that I assigned to @ernestotejeda Should we wait for his changes?

chienandalu committed:
[IMP] rma: teams flow

- If no RMA Team is set, we'll assign a default one to the new RMA.
- A sequence is now used to search for the top team and assign it.
- No default user is assigned when it's not in the context (i.e. portal
rmas).

chienandalu commented:
@pedrobaeza I've added a new commit fixing RMA Teams flows: https://github.com/OCA/rma/pull/160/commits/d4e316313a4989806fc2e6f935081fb3e9851658

pedrobaeza commented:
Are all the notified problems now solved?

chienandalu commented:
There are still two issues with Access errors in the portal views that I assigned to @ernestotejeda

pedrobaeza commented:
OK, let's merge this and let @ernestotejeda to open other PR.

/ocabot merge minor.
